lib/ueberauth/strategy/fitbit/oauth.ex

```diff
@@ -5,89 +5,73 @@ defmodule Ueberauth.Strategy.Fitbit.OAuth do
   Add `client_id` and `client_secret` to your configuration:
 
   config :ueberauth, Ueberauth.Strategy.Fitbit.OAuth,
-    client_id: System.get_env("FITBIT_APP_ID"),
-    client_secret: System.get_env("FITBIT_APP_SECRET")
+    client_id: System.get_env("FITBIT_CLIENT_ID"),
+    client_secret: System.get_env("FITBIT_CLIENT_SECRET"),
   """
   use OAuth2.Strategy
+  alias OAuth2.Strategy.AuthCode
 
-  @defaults [
+  @fitbit_config [
     strategy: __MODULE__,
-    site: "https://api.fitbit.com",
+    headers: [{"Content-Type", "application/x-www-form-urlencoded"}],
+    site: "https://api.fitbit.com/",
     authorize_url: "https://www.fitbit.com/oauth2/authorize",
-    token_url: "https://api.fitbit.com/oauth2/token",
-    headers: [{"Content-Type", "application/x-www-form-urlencoded"}]
+    token_url: "https://api.fitbit.com/oauth2/token"
   ]
 
-  @doc false
-  def options(opts \\ [], app \\ Application) do
-    config = app.get_env(:ueberauth, Ueberauth.Strategy.Fitbit.OAuth)
+  defp client(opts \\ []) do
+    opts =
+      @fitbit_config ++
+        oauth_config() ++
+        opts
 
-    @defaults
-    |> Keyword.merge(config)
-    |> Keyword.merge(opts)
+    opts
+    |> OAuth2.Client.new()
+    |> OAuth2.Client.put_serializer("application/json", Jason)
```

Comment:
We could replace `Jason` with `Ueberauth.json_library()` to use the consumer option.